test/e2e_source_map_test.ts

```diff
@@ -73,6 +73,69 @@ describe('source maps', () => {
         .to.equal('input2.ts', 'second input file');
   });
 
+  it('handles files in different directories', function() {
+    const sources = new Map<string, string>();
+    sources.set('a/b/input1.ts', `
+        class X { field: number; }
+        let x : string = 'a string';
+        let y : string = 'another string';
+        let z : string = x + y;`);
+
+    sources.set('a/c/input2.ts', `
+        class A { field: number; }
+        let a : string = 'third string';
+        let b : string = 'fourth rate';
+        let c : string = a + b;`);
+
+    // Run tsickle+TSC to convert inputs to Closure JS files.
+    const {compiledJS, sourceMap} = compile(sources, 'a/d/output.js');
+
+    const {line: stringXLine, column: stringXColumn} = getLineAndColumn(compiledJS, 'a string');
+    const {line: stringBLine, column: stringBColumn} = getLineAndColumn(compiledJS, 'fourth rate');
+
+    expect(sourceMap.originalPositionFor({line: stringXLine, column: stringXColumn}).line)
+        .to.equal(3, 'first string definition');
+    expect(sourceMap.originalPositionFor({line: stringXLine, column: stringXColumn}).source)
+        .to.equal('a/b/input1.ts', 'first input file');
+    expect(sourceMap.originalPositionFor({line: stringBLine, column: stringBColumn}).line)
+        .to.equal(4, 'fouth string definition');
+    expect(sourceMap.originalPositionFor({line: stringBLine, column: stringBColumn}).source)
+        .to.equal('a/c/input2.ts', 'second input file');
+  });
+
+  it('works when not tsickle processing some input', function() {
+    const sources = new Map<string, string>();
+    sources.set('input1.ts', `
+        /** @Annotation */
+        function classAnnotation(t: any) { return t; }
+
+        @classAnnotation
+        class DecoratorTest {
+          public methodName(s: string): string { return s; }
+        }`);
+
+    sources.set('input2.ts', `
+        class A { field: number; }
+        let a : string = 'third string';
+        let b : string = 'fourth rate';
+        let c : string = a + b;`);
+
+    // Run tsickle+TSC to convert inputs to Closure JS files.
+    const {compiledJS, sourceMap} = compile(sources, 'output.js', new Set<string>(['input2.ts']));
+
+    const {line: methodLine, column: methodColumn} = getLineAndColumn(compiledJS, 'methodName');
+    const {line: stringBLine, column: stringBColumn} = getLineAndColumn(compiledJS, 'fourth rate');
+
+    expect(sourceMap.originalPositionFor({line: methodLine, column: methodColumn}).line)
+        .to.equal(7, 'method definition');
+    expect(sourceMap.originalPositionFor({line: methodLine, column: methodColumn}).source)
```

Comment:
You're correct.  This is a separate bug that I discovered while working on this.  If you have some files getting decorator downleveled and others not, previously you'd get an undefined val exception when writing out because we'd try to compose decorator downlevel source maps with tsickle source maps when the decorator downlevel source maps weren't there.  I think this test would make more sense if I changed input2.ts to contain some annotations and change some of the comments.